Content.Server/_Lagrange/StationEvents/Events/DistressSignalRule.cs

```diff
@@ -0,0 +1,210 @@
+
+        // Initialize the Station component, giving us access to the randomly-generated name.
+        var shuttleStation = StationSystem.InitializeNewStation(shuttleProto.Stations[component.MapId], gridUids);
+        var metaData = MetaData((EntityUid) shuttleStation);
+        var shipNameParts = metaData.EntityName.Split(' ');
+        component.Designation = shipNameParts[^1];
+
+        // Announce the distress signal to the players.
+        var str = Loc.GetString("station-event-distress-signal-start-announcement",
+            ("designation", component.Designation),
+            ("x", Math.Round(offset.X)),
+            ("y", Math.Round(offset.Y))
+        );
+
+        ChatSystem.DispatchGlobalAnnouncement(str, colorOverride: Color.FromHex("#18abf5"));
+
+        // Send the grid to the announced location.
+        if (TryComp<ShuttleComponent>(gridUid, out var shuttle))
+        {
+            _shuttle.FTLTravel(gridUid, shuttle, location, 0, 0);
+        }
+    }
+
+    public void AddObjective(DistressSignalObjectiveComponent objective, DistressSignalRuleComponent component)
+    {
+        objective.DistressSignalRuleComponent = component;
+        component.Objectives.Add(objective);
+    }
+
+    protected override void ActiveTick(EntityUid uid, DistressSignalRuleComponent component, GameRuleComponent gameRule, float frameTime)
+    {
+        base.ActiveTick(uid, component, gameRule, frameTime);
+
+        if (component.TimerRunning)
+            return;
+
+        if (component.Objectives.Count == 0)
+        {
+            Log.Error($"Distress signal '{component.MapId}' lacks objectives and was thus aborted.");
+            if (component.GridUid is not null)
+                Del(component.GridUid);
+            GameTicker.EndGameRule(uid);
+            return;
+        }
+
+        // Determine whether a (partially) successful attempt to address the distress signal has been made.
+        int successCount = 0;
+        int failureCount = 0;
+
+        foreach (var objective in component.Objectives)
+        {
+            if (objective is null)
+            {
+                failureCount++;
+                continue;
+            }
+
+            if (objective.Failed)
+            {
+                if (objective.Critical)
+                {
+                    component.ObjectivesCompleted = false;
+                    GameTicker.EndGameRule(uid);
+                    return;
+                }
+
+                failureCount++;
+            }
+            else if (objective.Completed)
+            {
+                component.ObjectivesCompleted = true;
+                successCount++;
+            }
+        }
+
+        if (successCount + failureCount >= component.Objectives.Count)
+        {
+            component.ObjectivesCompleted = successCount > 0;
+
+            // Ensure the objectives *stay* completed.
+            component.TimerRunning = true;
+            Timer.Spawn(TimeSpan.FromSeconds(15), () =>
```

Comment:
You would know better than I, but is this worth keeping in config? Like, if the spawning process takes shorter or longer than 15s, can we adjust that per deploy? If it's constant, cool.